modules/telestion-api/src/main/java/de/wuespace/telestion/api/message/JsonMessage.java

```diff
@@ -1,141 +1,346 @@
 	 *
-	 * @return json representation of message
+	 * @param json the JSON {@link String} that contains the necessary information to construct
+	 *             the specified {@link JsonMessage}
+	 * @param type the class of the target {@link JsonMessage}
+	 * @param <T>  the type of the target {@link JsonMessage}
+	 * @return the decoded message
+	 * @throws DecodeException if the JSON string does not contain the necessary information to successfully
+	 *                         construct the specified {@link JsonMessage}
 	 */
-	default JsonObject json() {
+	static <T extends JsonMessage> T from(String json, Class<T> type) throws DecodeException {
+		return Json.CODEC.fromString(json, type);
+	}
+
+	/**
+	 * Constructs a {@link JsonMessage} from a plain {@link Object}.
+	 *
+	 * @param json the plain {@link Object} that contains the necessary information to construct
+	 *             the specified {@link JsonMessage}
+	 * @param type the class of the target {@link JsonMessage}
+	 * @param <T>  the type of the target {@link JsonMessage}
+	 * @return the decoded message
+	 * @throws DecodeException if the plain object does not contain the necessary information to successfully
+	 *                         construct the specified {@link JsonMessage}
+	 */
+	static <T extends JsonMessage> T from(Object json, Class<T> type) throws DecodeException {
+		return Json.CODEC.fromValue(json, type);
+	}
+
+	/**
+	 * Constructs a {@link JsonMessage} from a Vert.x EventBus {@link Message} body.
+	 *
+	 * @param message the message which body contains the necessary information to construct
+	 *                the specified {@link JsonMessage}
+	 * @param type    the class of the target {@link JsonMessage}
+	 * @param <T>     the type of the target {@link JsonMessage}
+	 * @return the decoded message
+	 * @throws DecodeException if the raw message body does not contain the necessary information to successfully
+	 *                         construct the specified {@link JsonMessage}
+	 */
+	static <T extends JsonMessage> T from(Message<?> message, Class<T> type) throws DecodeException {
+		return from(message.body(), type);
+	}
+
+	///
+	/// SYNCHRONOUS ENCODING SECTION
+	///
+
+	/**
+	 * @deprecated Use {@link #toJsonObject()} instead.
+	 */
+	@Deprecated(since = "0.9.0")
+	default JsonObject json() throws IllegalArgumentException {
+		return toJsonObject();
+	}
+
+	/**
+	 * Constructs a {@link JsonObject} from the {@link JsonMessage}.
+	 *
+	 * @return the constructed JSON object
+	 * @throws IllegalArgumentException if the JSON object cannot represent the type of
+	 *                                  any {@link JsonMessage} property
+	 */
+	default JsonObject toJsonObject() throws IllegalArgumentException {
 		return JsonObject.mapFrom(this);
 	}
+
+	/**
+	 * Constructs a {@link String} containing the properties of the {@link JsonMessage} as JSON values.
+	 *
+	 * @param pretty when {@code true} the JSON output is properly formatted
```

Comment:
```suggestion
	 * @param pretty if {@code true} the JSON output is properly formatted
```
I am not entirely certain, but I think here we need to use 'if' :)